support/config/deployment.go

```diff
@@ -286,11 +288,53 @@ func (c *DeploymentConfig) SetDefaults(hcp *hyperv1.HostedControlPlane, multiZon
 	}
 	c.DebugDeployments = debugDeployments(hcp)
 
+	c.ResourceRequestOverrides = resourceRequestOverrides(hcp)
+
 	c.setLocation(hcp, multiZoneSpreadLabels)
 	// TODO (alberto): make this private, atm is needed for the konnectivity agent daemonset.
 	c.SetReleaseImageAnnotation(hcp.Spec.ReleaseImage)
 }
 
+func resourceRequestOverrides(hcp *hyperv1.HostedControlPlane) ResourceOverrides {
```

Comment:
it'd be nice to have some log that shows the result to apply after parsing the input.